### DISCUSSION TITLE
Search functionality on labels? 👀

### DISCUSSION THREAD

**Discussion by zacrafidi** [flekschas/regl-scatterplot#121] [Q&A] [answered]
Hey there 👋

Not sure if this is a silly question but curious to know if regl-scatterplot has the ability to search/highlight scatter labels like in https://projector.tensorflow.org/ - or would that something need to be built?

(btw amazing work - repo/examples are awesome!)

Zac

**Comment by flekschas** [comment, marked as answer]
Unfortunately, there isn't any built in functionality for that as `regl-scatterplot` is a bit more "low level" than Tensorflow's Projector app.

Having said that, it should be fairly simple to implement this yourself depending on your data source and structure. For instance, you could index the point labels with something like a TRIE or DAWG (e.g., https://github.com/mckoss/dawg) for fuzzy string matching and then use `scatterplot.select(pointIndices)`.

If you happen to work in with Python, there's a sister application called [`jupyter-scatter`](https://github.com/flekschas/jupyter-scatter), which integrates into the Pandas ecosystem. That integration makes searching for data points fairly simple.

```py
import jscatter
scatter = jscatter.Scatter(data=df, x="column_a", y="column_b")
scatter.selection(df.query("label == 'my label'").index)
```

**Reply by zacrafidi** [reply]
Amazing thanks!

**Comment by flekschas** [comment]
Also, there are no silly questions, so keep them coming! :)
